File: RedSheet.DbEntities/Models/vProjectCultureRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectCultureRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectCultureRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int CountryId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int CultureCategoryId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int CultureId { get; set; } 
 
	[Required]
	public bool IsEgalitarian { get; set; } 
 
	[Required]
	public bool IsIndividualistic { get; set; } 
 
	[Required]
	public bool IsMonochronic { get; set; } 
 
	[Required]
	public bool IsShortTerm { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectCultureId { get; set; } 

	public vProjectCultureRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-project-culture-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectCultureRecord {
    constructor(vProjectCultureRecord?: vProjectCultureRecord )  {
        let properties = [ "countryId", "cultureCategoryId", "cultureId", "isEgalitarian", "isIndividualistic", "isMonochronic", "isShortTerm", "projectCultureId",];
        for (let property of properties)
            if (vProjectCultureRecord && vProjectCultureRecord[property])
                this[property] = vProjectCultureRecord[property];
    }
 
	countryId : number =   0 ;
 
    @range(1,2147483647)
	cultureCategoryId : number =   undefined;
 
    @range(1,2147483647)
	cultureId : number =   undefined;
 
	isEgalitarian : boolean = false ;
 
	isIndividualistic : boolean = false ;
 
	isMonochronic : boolean = false ;
 
	isShortTerm : boolean = false ;
 
    @range(1,2147483647)
	projectCultureId : number =   undefined;


}
